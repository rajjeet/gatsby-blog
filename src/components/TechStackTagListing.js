import React from "react";
import styled from 'styled-components';
import * as theme from '../utils/theme';

const TechStackTagListing = ({className, tags}) => (
    <div className={className}>
        Tech Tags:
        {
            tags.map(tag => (
                <span key={tag.label}>{tag.label}</span>
            ))
        }
    </div>
);

const StyledTechStackTagListing = styled(TechStackTagListing)`
  margin: .5em auto;
   span {
   background-color: ${theme.primaryColor};
    color: white;
    border-radius: 5px;
    font-size: .8em;
    font-weight: bolder;
    padding: .2em .5em;
    margin: .2em .4em;
   }
`;

export default StyledTechStackTagListing;