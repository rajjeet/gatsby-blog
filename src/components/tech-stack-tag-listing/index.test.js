import React from 'react';
import renderer from 'react-test-renderer';
import TechStackTagListing from './index';

const makeProps = () => ({
  className: {},
  tags: [],
});

describe('<TechStackTagListing />', () => {
  it('should render', () => {
    const tree = renderer.create(<TechStackTagListing {...makeProps()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
