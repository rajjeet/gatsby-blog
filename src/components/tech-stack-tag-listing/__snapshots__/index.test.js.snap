// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TechStackTagListing /> should render 1`] = `
<div
  className="[object Object] sc-bdVaJa kDwnaI"
>
  Tech Tags:
</div>
`;
